### PR TITLE
Updates for the VEW1D channel algorithm

### PR DESCRIPTION
**Pull request by shinbunya**
Updates for the VEW1D channel algorithm

These updates include functionality enhancements as well as bug fixes.

### CODE REVIEW

File: src/read_input.F [hunk 1]
```diff
@@ -199,7 +201,8 @@ SUBROUTINE READ_INPUT()
 C.... "new" formula for computing vel in wetting/drying routine
      &  slim, windlim, directvelWD, useHF
       NAMELIST /inundationOutputControl/ inundationOutput, inunThresh
-
+C.... SB
+      NAMELIST /channelWetPerimeterControl/ activateChannelWetPerimeter
```
Comment (zcobell) [inline src/read_input.F:205]: Same as above

File: thirdparty/swan/SwanReadADCGrid.ftn90 [hunk 2]
```diff
@@ -197,7 +197,10 @@ subroutine SwanReadADCGrid
 !NADC             vmark(ivert ) = vm
 !NADC             vmark(ivert1) = vm
 !ADC            vmark(find(node_dict,ivert )) = vm
-!ADC            vmark(find(node_dict,ivert1)) = vm
+!ADC            n3 = find(node_dict,ivert1)
```
Comment (zcobell) [inline thirdparty/swan/SwanReadADCGrid.ftn90:200]: Tagging @caseydietrich for awareness as he works on his implementation of a new version of SWAN

**Review by zcobell — CHANGES_REQUESTED** [review]
Hey @shinbunya - just a couple requested changes for clarity and one thing that looks like it was deleted in error

File: prep/read_global.F [hunk 3]
```diff
@@ -548,6 +548,9 @@ SUBROUTINE READ15()
      &  dynamicWaterLevelCorrectionRampStart, dynamicWaterLevelCorrectionRampEnd,
      &  dynamicWaterLevelCorrectionRampReferenceTime
 
+C.....Channel wet perimeter control 11/06/2023 sb
+      NAMELIST /channelWetPerimeterControl/ activateChannelWetPerimeter
```
[the remark this answers is not in the document]
Comment (shinbunya) [reply prep/read_global.F:552]: Would it be also reasonable if the namelist name `channelWetPerimeterControl` is swapped with `vew1DChannelControl` as this is not really relevant to wet/dry?
And this way the variable name `activateChannelWetPerimeter ` may stay as it is.

File: src/nodalattr.F [hunk 4]
```diff
@@ -2173,6 +2182,75 @@ END SUBROUTINE CalculateTimeVaryingTau0
...
+C     ----------------------------------------------------------------
+C     S U B R O U T I N E
+C     I N I T  C H A N N E L  W E T  P E R I M E T E R
+C     ----------------------------------------------------------------
+C
+C     Subroutine to initialize variable values for bottom friction
+C     evaluation considering wet perimeter  11/06/2023 sb
+C
+C     ----------------------------------------------------------------
+      SUBROUTINE initChannelWetPerimeter()
```
[the remark this answers is not in the document]
Comment (shinbunya) [reply src/nodalattr.F:2194]: Agreed.

File: src/nodalattr.F [hunk 5]
```diff
@@ -2250,42 +2328,21 @@ SUBROUTINE Apply2DBottomFriction(UU1, VV1, DP, ETA2, G,
...
-                      WCH = floor(ManningsN(I))
-                      ManN = ManningsN(I) - WCH
-                      FRIC(I)=g*((1.D0+2.D0*HT/WCH)**(2.D0/3.D0)*ManN)**2.d0
-     &                  /( HT**(1.d0/3.d0) )
-                      IF(FRIC(I).LT.BFCdLLimit) THEN
-                         FRIC(I) = BFCdLLimit
-                      ENDIF
-                   ELSE
-                      FRIC(I)=g*ManningsN(I)**2.d0
+                IF (ManningsN(I).le.0d0) cycle ! Can use Cd where man <= 0
```
[the remark this answers is not in the document]
Comment (shinbunya) [reply src/nodalattr.F:2331]: I'll check that. Thanks for catching this.

File: prep/read_global.F (hunk 3 above)
[the remark this answers is not in the document]
Comment (zcobell) [reply prep/read_global.F:552]: I think that works. Then we can use that namelist in the future for other VEW parameters too.

**Review by zcobell — APPROVED** [review]

**Pull request merged**